app.py

```diff
@@ -83,7 +90,13 @@ async def run_groups_authn():
         request = Request()
         creds.refresh(request)
 
-    # sync IAM groups to Cloud SQL instances
-    await groups_sync(iam_groups, sql_instances, creds, private_ip)
-
+    try:
+        # sync IAM groups to Cloud SQL instances
+        await groups_sync(iam_groups, sql_instances, creds, group_roles, private_ip)
+    except GroupRoleMaxLengthError as e:
```

Comment:
Can we throw this error before even calling group_sync?

Reply:
(it looks like probably not because it's different for postgres and mysql?)

Reply:
Correct, the fact that MySQL and Postgres have different character limits makes it more tricky. I need to first know the database type before I can validate the length of the role and throw the error.